src/main/java/interaction/rule/InteractionRule.java

```diff
@@ -0,0 +1,11 @@
+package interaction.rule;
+
+import action.Action;
+import object.GameObject;
+
+import java.util.Collection;
+
+public interface InteractionRule {
+
+    Collection<Action> getActions(GameObject firstObject, GameObject secondObject);
```

Comment:
```
Collection<Action> getActions(GameObject firstObject, GameObject secondObject);
```
equals to 
```
Collection<Action> getActions(Interaction interaction);
```

but semantically better and API oriented

create Interaction interface and refactor Interaction class to GameObjectInteraction